### PR TITLE
DAOS-15121 test: fix run_remote log without timeout

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag: pr DaosCoreTestDfuse
Skip-unit-tests: true
Skip-fault-injection-test: true

Fix bug in the log.debug call when there is not a timeout.

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by ashleypittman — APPROVED** [review]

File: src/tests/ftest/util/run_utils.py [hunk 1]
```diff
@@ -405,7 +405,7 @@ def run_remote(log, hosts, command, verbose=True, timeout=120, task_debug=False,
     task.set_info("ssh_options", "-oForwardAgent=yes")
     if verbose:
         if timeout is None:
-            log.debug("Running on %s without a timeout: %s", hosts, timeout, command)
+            log.debug("Running on %s without a timeout: %s", hosts, command)
```
Comment (phender) [inline src/tests/ftest/util/run_utils.py:408]: Thoughts on adding a `test_harness_unit_run_remote_no_timeout` test case to `harness/unit.py` to verify this arguably simple change?

Comment (ashleypittman) [reply src/tests/ftest/util/run_utils.py:408]: I'd sooner ask why are we allowing commands without timeouts.

Comment (daltonbohning) [reply src/tests/ftest/util/run_utils.py:408]: The default timeout is 120, so a test would have to explicitly pass `timeout=None`. And the entire test itself is still bound by a timeout. Most cases probably don't need `timeout=None`, but I think it's okay to support it.

Comment (daltonbohning) [reply src/tests/ftest/util/run_utils.py:408]: That said, maybe tests should just adapt to always having a timeout

Comment (daltonbohning) [reply src/tests/ftest/util/run_utils.py:408]: > test_harness_unit_run_remote_no_timeout

I think that's overkill because we should trip over the error this PR fixes if `verbose=True` and `timeout is None`. And those don't tend to be dynamic for an individual test. I suspect something subtle changed somewhere for this test to trip over this now.

Comment (ashleypittman) [reply src/tests/ftest/util/run_utils.py:408]: > That said, maybe tests should just adapt to always having a timeout

If the run_remote() timeout times out then that's a test failure, if the overall test times out then that's a test error.  It's a minor point but there's no case for passing in timeout=None as a timeout value here.

Comment (daltonbohning) [reply src/tests/ftest/util/run_utils.py:408]: > there's no case for passing in timeout=None as a timeout value here

It appears to just be used for legacy cases where commands just didn't specify a timeout in the past. E.g. in `fio_utils` the timeout is defaulted to `None`. There isn't a sensible default here because adjusting the fio command options could very easily cause the command to run for a long time. But I think it's better for tests to just specify a timeout.

In short, I agree with you. But it would take a little digging to ensure we catch every case

Comment (ashleypittman) [reply src/tests/ftest/util/run_utils.py:408]: Let's not waste time adding additional test harnesses for configurations that should be dis-allowed then.

**Comment by daltonbohning** [comment]
Closed in favor of #13743
